Action/src/Action.java

```diff
@@ -0,0 +1,109 @@
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        System.out.println("The coffee machine has:");
+        System.out.println(countOfWater + " of water");
+        System.out.println(countOfMilk + " of milk");
+        System.out.println(countOfCoffeeBeans + " of coffee beans");
+        System.out.println(countOfMoney + " of money");
+        System.out.println("Write action (buy, fill, take):");
+        System.out.print(">");
+        String action = reader.readLine();
+
+        if(action.equals("buy"))
+            buy(countOfWater,countOfMilk,countOfCoffeeBeans,countOfMoney,countOfCups);
+        if(action.equals("fill"))
+            fill(countOfWater,countOfMilk,countOfCoffeeBeans,countOfMoney,countOfCups);
+        if(action.equals("take"))
+            take(countOfWater,countOfMilk,countOfCoffeeBeans,countOfMoney,countOfCups);
+
+        reader.close();
+
+    }
+    public static void buy(int countOfWater, int countOfMilk,int countOfCoffeeBeans, int countOfMoney,int countOfCups) throws IOException{
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino:");
+        System.out.print(">");
+        String number = reader.readLine();
+        System.out.println(number);
+        if(number.equals("1")){
+            System.out.println("The coffee machine has:");
+            System.out.println(countOfWater - 250 + " of water");
+            System.out.println(countOfMilk + " of milk");
+            System.out.println(countOfCoffeeBeans - 16 + " of coffee beans");
+            System.out.println(countOfCups - 1 + " of disposable cups");
+            System.out.println(countOfMoney + 4 + " of money");
+        }
+        if(number.equals("2")){
+            System.out.println("The coffee machine has:");
+            System.out.println(countOfWater - 350 + " of water");
+            System.out.println(countOfMilk - 75 + " of milk");
+            System.out.println(countOfCoffeeBeans - 20 + " of coffee beans");
+            System.out.println(countOfCups - 1 + " of disposable cups");
+            System.out.println(countOfMoney + 7 + " of money");
+        }
+
+        if(number.equals("3")){
+            System.out.println("The coffee machine has:");
+            System.out.println(countOfWater - 200 + " of water");
+            System.out.println(countOfMilk - 100 + " of milk");
+            System.out.println(countOfCoffeeBeans - 12 + " of coffee beans");
+            System.out.println(countOfCups - 1 + " of disposable cups");
+            System.out.println(countOfMoney + 6 + " of money");
+        }
+        reader.close();
+
+    }
+
+    public static void fill(int countOfWater, int countOfMilk,int countOfCoffeeBeans, int countOfMoney,int countOfCups) throws IOException{
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
```

Comment:
@AraChilingaryan method is too big , also why you are throwing an exception ?

Reply:
because i used Bufferedreader

Reply:
or i can catch it..but i thought that throwing or catching doesnt have big differences in this leve;

Reply:
done